code/modules/mob/dead/observer/observer.dm

```diff
@@ -758,6 +766,37 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		to_chat(src, span_notice("Data HUDs enabled."))
 		data_huds_on = 1
 
+/mob/dead/observer/verb/unlock_antag_sight()
+	set category = "Ghost"
+	set name = "Unlock Antag HUD"
+	if(!client)
+		return
+	if(CONFIG_GET(flag/disable_ghost_unlock_antag_hud))
+		to_chat(src, span_warning("Unlocking the Antag HUD is currently disabled."))
+		return
+	if(antag_sight_unlocked)
+		to_chat(src, span_warning("Antag HUD already unlocked."))
+		return
+
+	var/response = tgui_alert(src, "Are you sure you want to reveal antagonists? [can_reenter_corpse ? "You will be unrevivable, and" : "You will be"] disqualified from ghost role offers and spawners.", "Confirm Antag HUD Unlock", list("Yes - I am sure", "No - Cancel"))
+	if(response != "Yes - I am sure")
+		return
+
+	antag_sight_unlocked = TRUE
+
+	if(can_reenter_corpse) // force DNR, copy pasta of that verb
```

Comment:
Maybe you shouldn't copy and paste the code.

Reply:
Again just requires another proc to abstract out that logic from inside of the DNR verb.